hash.c

```diff
@@ -6842,6 +6842,138 @@ env_update(VALUE env, VALUE hash)
  *    synonyms.keys # => [:hello, :world]
  *
  *  Note that setting the default proc will clear the default value and vice versa.
+ *
+ *  === What's Here
+ *
+ *  \Class \Hash provides methods that are useful for:
+ *
+ *  - {Creating a Hash}[#class-Hash-label-Methods+for+Creating+a+Hash]
+ *  - {Setting Hash State}[#class-Hash-label-Methods+for+Setting+Hash+State]
+ *  - {Querying}[#class-Hash-label-Methods+for+Querying]
+ *  - {Comparing}[#class-Hash-label-Methods+for+Comparing]
+ *  - {Fetching}[#class-Hash-label-Methods+for+Fetching]
+ *  - {Assigning}[#class-Hash-label-Methods+for+Assigning]
+ *  - {Deleting}[#class-Hash-label-Methods+for+Deleting]
+ *  - {Iterating}[#class-Hash-label-Methods+for+Iterating]
+ *  - {Converting}[#class-Hash-label-Methods+for+Converting]
+ *  - {Transforming Keys and Values}[#class-Hash-label-Methods+for+Transforming+Keys+and+Values]
+ *  - {And more....}[#class-Hash-label-Other+Methods]
+ *
+ *  \Class \Hash also includes methods from module Enumerable.
+ *
+ *  ==== Methods for Creating a \Hash
+ *
+ *  ::[]:: Returns a new hash populated with given objects.
+ *  ::new:: Returns a new empty hash.
+ *  ::try_convert:: Returns a new hash created from a given object.
+ *
+ *  ==== Methods for Setting \Hash State
+ *
+ *  #compare_by_identity:: Sets +self+ to consider only identity in comparing keys.
+ *  #default=:: Sets the default to a given value.
+ *  #default_proc=:: Sets the default proc to a given proc.
+ *  #rehash:: Rebuilds the hash table by recomputing the hash index for each key.
+ *
+ *  ==== Methods for Querying
+ *
+ *  #any?:: Returns +true+ if any element satisfies a given criterion; +false+ otherwise.
+ *  #compare_by_identity?:: Returns +true+ if compare_by_identity has been called, +false+ otherwise.
+ *  #default:: Returns the default value, or the default value for a given key.
+ *  #default_proc:: Returns the default proc.
+ *  #empty?:: Returns +true+ if there are no entries, +false+ otherwise.
+ *  #eql?:: Returns +true+ if a given object is equal to +self+, +false+ otherwise.
+ *  #hash:: Returns the integer hash code.
+ *  #has_key?:: Returns +true+ if a given object is a key in +self+, otherwise +false+.
+ *  #has_value?:: Returns +true+ if a given object is a value in +self+, otherwise +false+.
+ *  #include?:: Returns +true+ if a given object is a key in +self+, otherwise +false+.
+ *  #key?:: Returns +true+ if a given object is a key in +self+, otherwise +false+.
```

Comment:
Instead of listing aliases as separate lines, use `#key?, #has_key?, #include?, #member?::`.  Similar for other method aliases in this section and other sections.

Reply:
Done.